subsys/bluetooth/common/dummy.c

```diff
@@ -59,3 +59,17 @@ BUILD_ASSERT(!IS_ENABLED(CONFIG_LOG_MODE_IMMEDIATE), "Immediate logging "
 	     "on selected backend(s) not "
 	     "supported with the software Link Layer");
 #endif
+
+#if defined(CONFIG_BT_CONN) && defined(CONFIG_BT_HCI_ACL_FLOW_CONTROL)
+/* When ACL flow-control is enabled, the host needs more ACL buffers than
+ * maximum ACL links. This is because of the way we re-assemble ACL packets into
+ * L2CAP PDUs.
+ *
+ * We keep around the first buffer (that comes from the driver) to do
+ * re-assembly into, and if all links are re-assembling, there will be no buffer
+ * available for the HCI driver to allocate from.
+ *
+ * Fixing it properly involves a re-design of the HCI driver interface.
+ */
+BUILD_ASSERT(CONFIG_BT_BUF_ACL_RX_COUNT > CONFIG_BT_MAX_CONN);
+#endif
```

Comment:
Was wondering if we could enforce this in Kconfig, but looks like it cannot do arithmetic, but we could do 

```c
config BT_BUF_ACL_RX_COUNT
	int "Number of incoming ACL data buffers"
	default BT_MAX_CONN if BT_HCI_ACL_FLOW_CONTROL
	default 6
	range BT_MAX_CONN 64 if BT_HCI_ACL_FLOW_CONTROL
	range 1 64
```

and then instantiate 1 more buffer than `BT_BUF_ACL_RX_COUNT` to add the final requirement, but this approach isn't really clean either. 

I just don't like having a Kconfig requirement that triggers a build assert (without a message) for a Kconfig option, rather than adding the minimum values in the Kconfig option itself, but I guess we don't really have a choice. 

We should consider putting the comment into the build assert though, so that it becomes visible if improperly configured by an application, rather than users having to attempt to find this requirement in a file called `dummy.c`.